lib/newProfile.ml

```diff
@@ -176,13 +176,27 @@ let leave_sums ?time name () =
    accu. sums <- (start', next) :: rest;
     sum, dur
 
+let rec pptime fmt t = let open Format in function
+  | [] -> assert false
+  | [unit] ->
+    if t >= 1. then fprintf fmt "%03.0f%s" t unit
+    else fprintf fmt "0%s" unit
+  | unit :: nextunit :: rest ->
+    (* float t is time in [unit] *)
+    if t >= 1. then fprintf fmt "%03.0f%s %03.0f%s" t unit (Float.rem (t *. 1_000.) 1_000.) nextunit
```

Comment:
not sure if we want to 0-pad here, I feel it's slightly easier to see the difference between 

023ms 234us
345ms 013us

rather than

23ms 234us
345ms 123us

but maybe it's just noisy